results/energy.tex

```diff
@@ -0,0 +1,142 @@
+The \emph{O0} levels were only evaluated with debugging symbols.
+
+\subsubsection{Results}
+
+In this section we look at the results of the energy consumption measurement.
+We first look at the whole picture by presenting all the measurements produced by the set of possible parameters.
+Later we will consider the best configured {\C} application together with best configured {\rust} application, for each workload.
+
+\begin{figure}[H]
+
+  \centering
+  \begin{subfigure}{0.49\textwidth}
+    \includegraphics[width=\textwidth]{results/plots/energy/25.png}
+    \caption{25ms}
+    \label{}
+  \end{subfigure}
+  \hfill
+  \begin{subfigure}{0.49\textwidth}
+    \includegraphics[width=\textwidth]{results/plots/energy/50.png}
+    \caption{50ms}
+    \label{}
+  \end{subfigure}
+
+    \begin{subfigure}{0.49\textwidth}
+    \includegraphics[width=\textwidth]{results/plots/energy/100.png}
+    \caption{100ms}
+    \label{}
+  \end{subfigure}
+  \hfill
+  \begin{subfigure}{0.49\textwidth}
+    \includegraphics[width=\textwidth]{results/plots/energy/500.png}
+    \caption{500ms}
+    \label{}
+  \end{subfigure}
+
+  \begin{subfigure}{0.49\textwidth}
+    \includegraphics[width=\textwidth]{results/plots/energy/1000.png}
+    \caption{1000ms}
+    \label{}
+  \end{subfigure}
+
+  \caption{{\rust} vs {\C} comparison for each workload}
+  \label{fig:res:energy:comparison}
+\end{figure}
+
+\autoref{fig:res:energy:comparison} shows the energy consumption for each of the workloads and compares the optimization levels for {\rust} and {\C}.
+We see that {\rust} and {\C} has comparable consumption in each of the workload configurations.
+It is also evident that an external bias, investigated below, accounts for the variance between the various optimization levels.
+
+Now we look at the relative performance comparing the version with lowest energy consumption for each workload.
+For the {\C} code we can easily see from \autoref{fig:res:energy:comparison} that this is produced by the \emph{Os} level.
+The {\rust} versions on the other hand are either \emph{O1}, \emph{O2} or \emph{O3}, although the difference is never larger that 1\%.
+
+\begin{figure}[H]
+  \includegraphics[width=\textwidth]{results/plots/energy/best.png}
+  \caption{{\rust} vs {\C} relative comparisons for best builds}
+  \label{fig:res:energy:best}
+\end{figure}
+
+\autoref{fig:res:energy:best} shows the relative energy consumption by setting the consumption of the {\C} \emph{Os} build to 1 for each of the workload.
+The {\rust} line plots the best performing {\rust} builds relative to this {\C} build.
+For reference the results for the \emph{O3} optimization level of the {\C} build is included.
+We see that the {\rust} code is always within the 15\% of the {\C} code in energy consumption and below the \emph{O3} line.
+
+
+The variance between the different optimization levels in \autoref{fig:res:energy:comparison} was unexpected.
```

Comment:
Kanskje gjøre klart at variance in the C code var veldig varierende? Rust sin er ganske stabil